test/e2e_node/container_manager_test.go

```diff
@@ -192,6 +195,9 @@ var _ = SIGDescribe("Container Manager Misc [Serial]", func() {
 						},
 					},
 				})
+				if err := e2epod.WaitForPodCondition(f.ClientSet, f.Namespace.Name, podName, "Ready", 120*time.Second, testutils.PodRunningReady); err != nil {
```

Comment:
I don't think this is it. In the failure output, the pod is getting rejected because there aren't enough resources (CPU) left on the node, not due to a race condition on checking Ready status.

For some reason we have system pods requesting too much CPU which occasionally interferes with this test.

Reply:
Yea, the restart-test-pod was rejected because it requested 950m cpu, but the node has 1000m cpu, and other 100m was occupied by the guaranteed-pod of the container-manager-test-case(see the code https://github.com/kubernetes/kubernetes/blob/324ba1324b4b10ae7ef4d9d0a0654da7b118b920/test/e2e_node/container_manager_test.go#L187) which I analysed in the issue,  the kubelet log can prove this inference:
`I0316 21:25:06.552829  172062 predicate.go:130] "Predicate failed on Pod" pod="restart-test-721/test-179d535e-98f2-4a77-9a56-27fd9c16901e" err="Node didn't have enough resource: cpu, requested: 950, used: 100, capacity: 1000"`

Reply:
Adding the pod ready status checking is because the code https://github.com/kubernetes/kubernetes/blob/324ba1324b4b10ae7ef4d9d0a0654da7b118b920/test/e2e_node/container_manager_test.go#L200 which checks the pid of nginx can pass even while the pod has not been created. see 


> ```
> I0316 21:25:08.688468  172062 kuberuntime_manager.go:889] "Creating container in pod" containerType="container" container="&Container{Name:guaranteede9d81584-c234-43a2-8e2f-12b067d05eec,Image:k8s.gcr.io/e2e-test-images/nginx:1.14-2,Command:[],Args:[],WorkingDir:,Ports:[]ContainerPort{},Env:[]EnvVar{},Resources:ResourceRequirements{Limits:ResourceList{cpu: {{100 -3} {<nil>} 100m DecimalSI},memory: {{52428800 0} {<nil>} 50Mi BinarySI},},Requests:ResourceList{cpu: {{100 -3} {<nil>} 100m DecimalSI},memory: {{52428800 0} {<nil>} 50Mi BinarySI},},},VolumeMounts:[]VolumeMount{},LivenessProbe:nil,ReadinessProbe:nil,Lifecycle:nil,TerminationMessagePath:/dev/termination-log,ImagePullPolicy:Never,SecurityContext:nil,Stdin:false,StdinOnce:false,TTY:false,EnvFrom:[]EnvFromSource{},TerminationMessagePolicy:File,VolumeDevices:[]VolumeDevice{},StartupProbe:nil,}" pod="kubelet-container-manager-8210/guaranteede9d81584-c234-43a2-8e2f-12b067d05eec"
> I0316 21:25:08.696208  172062 kubelet_pods.go:161] "Creating hosts mount for container" pod="kubelet-container-manager-8210/guaranteede9d81584-c234-43a2-8e2f-12b067d05eec" containerName="guaranteede9d81584-c234-43a2-8e2f-12b067d05eec" podIPs=[10.85.0.111 1100:200::6f] path=true
> ```
> 
> Since at 0316 21:25:08, the container of pod guaranteede9d81584 has just been creating, but the test case [sig-node] Container Manager Misc [Serial] has already finished at 0316 21:25:06, so the test case check condition may be not strict
> 
> ```
> W0316 21:43:30.267] [sig-node] Container Manager Misc [Serial] Validate OOM score adjustments [NodeFeature:OOMScoreAdj] once the node is setup 
> W0316 21:43:30.267]   guaranteed container's oom-score-adj should be -998
> W0316 21:43:30.267]   test/e2e_node/container_manager_test.go:173
> W0316 21:43:30.267] [BeforeEach] [sig-node] Container Manager Misc [Serial]
> W0316 21:43:30.267]   test/e2e/framework/framework.go:185
> W0316 21:43:30.267] STEP: Creating a kubernetes client
> W0316 21:43:30.268] STEP: Building a namespace api object, basename kubelet-container-manager
> W0316 21:43:30.268] Mar 16 21:25:06.470: INFO: Skipping waiting for service account
> W0316 21:43:30.268] [It] guaranteed container's oom-score-adj should be -998
> W0316 21:43:30.268]   test/e2e_node/container_manager_test.go:173
> W0316 21:43:30.268] [AfterEach] [sig-node] Container Manager Misc [Serial]
> W0316 21:43:30.268]   test/e2e/framework/framework.go:186
> W0316 21:43:30.268] Mar 16 21:25:06.507: INFO: Waiting up to 3m0s for all (but 0) nodes to be ready
> W0316 21:43:30.268] STEP: Destroying namespace "kubelet-container-manager-8210" for this suite.
> W0316 21:43:30.268] •
> W0316 21:43:30.269] ------------------------------
> W0316 21:43:30.269] [sig-node] Restart [Serial] [Slow] [Disruptive] Kubelet 
> ```